app/menus/index.js

```diff
@@ -248,6 +249,95 @@ class Menus {
     const script = `document.querySelectorAll('video').forEach(v => {v.removeAttribute("disablepictureinpicture"); v.toggleAttribute("controls");})`;
     this.window.webContents.executeJavaScript(script, true);
   }
+
+  joinMeeting() {
+    try {
+      const clipboardText = clipboard.readText();
+      if (this.isValidTeamsMeetingUrl(clipboardText)) {
+        this.joinMeetingWithUrl(clipboardText);
+        return;
+      }
+    } catch (error) {
+      console.error('Error checking clipboard:', error);
+    }
+    
+    this.startClipboardMonitoring();
+  }
+
+  startClipboardMonitoring() {
+    if (this.clipboardMonitoring?.interval) {
+      return; // Already monitoring
+    }
+
+    this.clipboardMonitoring = {
+      interval: null,
+      lastUrl: null
+    };
+
+    this.clipboardMonitoring.interval = setInterval(() => {
+      this.checkClipboardForMeetingUrl();
+    }, 1000);
+
+    this.showWaitingForUrlDialog();
+  }
+
+  checkClipboardForMeetingUrl() {
+    try {
+      const clipboardText = clipboard.readText();
+      const isValidUrl = this.isValidTeamsMeetingUrl(clipboardText);
+      
+      if (isValidUrl && clipboardText !== this.clipboardMonitoring.lastUrl) {
+        this.clipboardMonitoring.lastUrl = clipboardText;
+        this.joinMeetingWithUrl(clipboardText);
+        this.stopClipboardMonitoring();
+      }
+    } catch (error) {
+      console.error('Error checking clipboard:', error);
+    }
+  }
+
+  isValidTeamsMeetingUrl(text) {
+    if (typeof text !== 'string') {
+      return false;
+    }
+    const teamsUrlPattern = /^https:\/\/teams\.microsoft\.com\/l\/meetup-join\//;
```

Comment:
I think there might be other patterns as some teams uses teams.live.com and others... but we can always iterate and support other links and/or flows. More a future note for improvements.